pkg/api/controller.go

```diff
@@ -299,8 +303,11 @@ func (c *Controller) InitImageStore(reloadCtx context.Context) error {
 				if storageConfig.StorageDriver == nil {
 					// false positive lint - linter does not implement Lint method
 					// nolint: typecheck
-					subImageStore[route] = storage.NewImageStore(storageConfig.RootDirectory,
-						storageConfig.GC, storageConfig.GCDelay, storageConfig.Dedupe, storageConfig.Commit, c.Log, c.Metrics, linter)
+					_, ok := imgStoreMap[storageConfig.RootDirectory]
```

Comment:
We need filesystem equality check and not string equality here.

SameFile (fi1, fi2 FileInfo)
^ use this.